src/main/java/network/brightspots/rcv/CastVoteRecord.java

```diff
@@ -76,6 +76,25 @@ class CastVoteRecord {
     this.candidateRankings = new CandidateRankingsList(rankings);
   }
 
+  CastVoteRecord(
+          String contestId,
+          String tabulatorId,
+          String batchId,
+          String suppliedId,
+          String computedId,
+          String precinct,
+          String precinctPortion,
+          List<Pair<Integer, String>> rankings) {
+    this.contestId = contestId;
+    this.tabulatorId = tabulatorId;
+    this.batchId = batchId;
+    this.computedId = computedId;
+    this.suppliedId = suppliedId;
+    this.precinct = precinct;
+    this.precinctPortion = precinctPortion;
+    this.candidateRankings = new CandidateRankingsList(rankings);
+  }
```

Comment:
Even if you keep `computedId` optional, you can consolidate your logic a bit by having the constructors with fewer arguments call `this` with the full arg list instead of having all the redundant assignments across the different constructors.

Reply:
I refactored this with the newest commit [b281aa2](https://github.com/BrightSpots/rcv/pull/805/commits/b281aa2eeba86d3031dda8454fcdb13ce139cacb)